From 4823e7a999033f12969cbc26c96e5033b67d98d0 Mon Sep 17 00:00:00 2001
From: "Zhao, Xinda" <xinda.zhao@intel.com>
Date: Wed, 10 May 2017 15:52:14 +0800
Subject: [PATCH 2/3] Add GPU cgroup support when compiling docker

Correct runc and cli location that has GPU support when compiling docker

Signed-off-by: Zhao, Xinda <xinda.zhao@intel.com>
---
 hack/dockerfile/binaries-commits    | 2 +-
 hack/dockerfile/install-binaries.sh | 6 +++---
 hack/make.sh                        | 1 +
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/hack/dockerfile/binaries-commits b/hack/dockerfile/binaries-commits
index 4d16a49..e4e7740 100644
--- a/hack/dockerfile/binaries-commits
+++ b/hack/dockerfile/binaries-commits
@@ -11,5 +11,5 @@ VNDR_COMMIT=c56e082291115e369f77601f9c071dd0b87c7120
 BINDATA_COMMIT=a0ff2567cfb70903282db057e799fd826784d41d
 
 # CLI
-DOCKERCLI_REPO=https://github.com/docker/cli
+DOCKERCLI_REPO=https://github.com/zhenyw/cli.git
 DOCKERCLI_COMMIT=c3648a9c9400d45524cc71b8fca4085b192c626f
diff --git a/hack/dockerfile/install-binaries.sh b/hack/dockerfile/install-binaries.sh
index e7ad9be..2b24e2f 100755
--- a/hack/dockerfile/install-binaries.sh
+++ b/hack/dockerfile/install-binaries.sh
@@ -20,9 +20,9 @@ RUNC_BUILDTAGS="${RUNC_BUILDTAGS:-"seccomp apparmor selinux"}"
 
 install_runc() {
 	echo "Install runc version $RUNC_COMMIT"
-	git clone https://github.com/docker/runc.git "$GOPATH/src/github.com/opencontainers/runc"
+	git clone https://github.com/zhenyw/runc.git "$GOPATH/src/github.com/opencontainers/runc"
 	cd "$GOPATH/src/github.com/opencontainers/runc"
-	git checkout -q "$RUNC_COMMIT"
+	git checkout docker-runc
 	make BUILDTAGS="$RUNC_BUILDTAGS" $1
 	cp runc /usr/local/bin/docker-runc
 }
@@ -58,7 +58,7 @@ install_dockercli() {
 	echo "Install docker/cli version $DOCKERCLI_COMMIT"
 	git clone "$DOCKERCLI_REPO" "$GOPATH/src/github.com/docker/cli"
 	cd "$GOPATH/src/github.com/docker/cli"
-	git checkout -q "$DOCKERCLI_COMMIT"
+	git checkout docker-cli
 	go build -o /usr/local/bin/docker github.com/docker/cli/cmd/docker
 }
 
diff --git a/hack/make.sh b/hack/make.sh
index b8b04ed..ec7147b 100755
--- a/hack/make.sh
+++ b/hack/make.sh
@@ -235,6 +235,7 @@ copy_binaries() {
 	if [ "$(go env GOOS)/$(go env GOARCH)" == "$(go env GOHOSTOS)/$(go env GOHOSTARCH)" ]; then
 		if [ -x /usr/local/bin/docker-runc ]; then
 			echo "Copying nested executables into $dir"
+			cp -f `which "docker"` "$dir/"
 			for file in containerd containerd-shim containerd-ctr runc init proxy; do
 				cp -f `which "docker-$file"` "$dir/"
 				if [ "$2" == "hash" ]; then
-- 
2.7.4

